### DISCUSSION TITLE
How to save additional data (like label and layout) in zod object schema?

### DISCUSSION THREAD

**Discussion by bluepuma77** [colinhacks/zod#2866] [Q&A] [answered]
Is there a way to save additional data in a zod object schema?

Lets take the simple user example:
```ts
const User = z.object({
  username: z.string(),
  firstname: z.string(),
  lastname: z.string(),
});
```
And I would like to add a `label` and `class` information to each field, to be able to automatically create a layout-ed form with inputs from the schema.

**Comment by JacobWeisenburger** [comment, marked as answer]
> Is there a way to save additional data in a zod object schema?

Not really. It's not really the purpose of zod to do this.
